service/src/main/java/uk/gov/hmcts/reform/fpl/service/DraftCMOService.java

```diff
@@ -81,20 +98,36 @@ public DraftCMOService(DateFormatterService dateFormatterService,
 
     public CaseManagementOrder prepareCMO(Map<String, Object> caseData) {
         DynamicList list = mapper.convertValue(caseData.get("cmoHearingDateList"), DynamicList.class);
+
+        String hearingDate = null;
+        UUID id = null;
```

Comment:
[-0.5] Probably that could be abstracted into Element<String> and use optional. maps instead of ugly nulls and ifs.